Source/WebKit/UIProcess/FrameLoadState.cpp

```diff
@@ -113,7 +134,17 @@ void FrameLoadState::didFailLoad()
 void FrameLoadState::didSameDocumentNotification(const URL& url)
 {
     ASSERT(!url.isNull());
-    m_url = url.isNull() ? aboutBlankURL() : url;
+    setURL(url.isNull() ? aboutBlankURL() : url);
+}
+
+void FrameLoadState::setURL(const URL& url)
+{
+    m_url = url;
```

Comment:
Was m_url ever set somewhere else before than in didSameDocumentNotification(). If not, we will now make it possible to set m_url to null which was not possible earlier. That's because you left the null check in the caller. We should make sure that's right.